pkg/monitoring/metrics/virt-handler/metrics.go

```diff
@@ -42,7 +43,14 @@ func SetupMetrics(virtShareDir, nodeName string, MaxRequestsInFlight int, vmiInf
 	SetVersionInfo()
 
 	domainstats.SetupDomainStatsCollector(virtShareDir, nodeName, MaxRequestsInFlight, vmiInformer)
-	return operatormetrics.RegisterCollector(domainstats.Collector)
+
+	if vmiInformer != nil {
+		if err := migrationdomainstats.SetupMigrationStatsCollector(vmiInformer); err != nil {
+			return err
+		}
+	}
```

Comment:
I would prefer to leave this check to the `SetupMigrationStatsCollector` function:
```go
func SetupMigrationStatsCollector(vmiInformer cache.SharedIndexInformer) error {
	if vmiInformer == nil {
		return nil
	}

	var err error
	migrationdomainstatsHandler, err = newHandler(vmiInformer)
	return err
}
```
and here simpli call it:
```suggestion
	if err := migrationdomainstats.SetupMigrationStatsCollector(vmiInformer); err != nil {
		return err
	}
```